src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -287,6 +302,39 @@ int IVsWindowFrameNotify.OnSize()
             int IVsWindowFrameNotify2.OnClose(ref uint pgrfSaveOptions)
                 => Disconnect();
 
+            private int TryInitializeTextBuffer()
+            {
+                _documentTracker.AssertIsForeground();
+
+                if (TextBuffer is not null)
+                    return VSConstants.S_OK;
+
+                if (ErrorHandler.Succeeded(Frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocCookie, out var boxedDocCookie)) && boxedDocCookie is uint docCookie)
+                {
+                    var flags = (_VSRDTFLAGS)_runningDocumentTable.GetDocumentFlags(docCookie);
+                    if ((flags & (_VSRDTFLAGS)_VSRDTFLAGS4.RDT_PendingInitialization) != 0)
+                    {
+                        // This document is not yet initialized. Defer initialization to the next OnShow event.
+                        return VSConstants.S_OK;
+                    }
+                }
+
+                if (ErrorHandler.Succeeded(Frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocData, out var docData)))
+                {
+                    if (docData is IVsTextBuffer bufferAdapter)
+                    {
+                        TextBuffer = _documentTracker._editorAdaptersFactoryService.GetDocumentBuffer(bufferAdapter);
```

Comment:
Just want to make sure, is our cleanup code resilient to this not being her

Reply:
Most of the stack here is null annotated and everything I saw had null checks in place. To outside code, this just shows up as the `DocumentId` being null, which is also the case any time a non-Roslyn document has focus.